src/installer/image/unpack.go

```diff
@@ -3,84 +3,154 @@ package image
 import (
 	"context"
 	"fmt"
+	"path"
 	"path/filepath"
 
-	"github.com/containers/image/v5/copy"
+	"github.com/Dynatrace/dynatrace-operator/src/dockerkeychain"
+	"github.com/Dynatrace/dynatrace-operator/src/installer/common"
 	"github.com/containers/image/v5/manifest"
 	"github.com/containers/image/v5/signature"
-	"github.com/containers/image/v5/types"
-	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/pkg/errors"
 	"github.com/spf13/afero"
 )
 
+const (
+	// MediaTypeImageLayerGzip is the media type used for gzipped layers
+	// referenced by the manifest.
+	mediaTypeImageLayerGzip = "application/vnd.oci.image.layer.v1.tar+gzip"
+
+	// MediaTypeImageLayerZstd is the media type used for zstd compressed
+	// layers referenced by the manifest.
+	mediaTypeImageLayerZstd = "application/vnd.oci.image.layer.v1.tar+zstd"
+
+	mediaTypeImageLayerDockerRootFs = "application/vnd.docker.image.rootfs.diff.tar.gzip"
+)
+
 type imagePullInfo struct {
-	imageCacheDir  string
-	targetDir      string
-	sourceCtx      *types.SystemContext
-	destinationCtx *types.SystemContext
-	sourceRef      *types.ImageReference
-	destinationRef *types.ImageReference
+	imageCacheDir string
+	targetDir     string
 }
 
-func (installer Installer) extractAgentBinariesFromImage(pullInfo imagePullInfo) error {
-	manifestBlob, err := copyImageToCache(pullInfo)
+func (installer Installer) extractAgentBinariesFromImage(pullInfo imagePullInfo, registryAuthPath string, imageName string) error { //nolint
+	img, err := installer.pullImageInfo(registryAuthPath, imageName)
 	if err != nil {
-		log.Info("failed to get manifests blob",
-			"image", installer.props.ImageUri,
-		)
+		log.Info("pullImageInfo", "error", err)
+		return err
+	}
+
+	image := *img
+
+	imgManifest, err := image.Manifest()
+	if err != nil {
+		log.Info("imgManifest", "error", err)
+		return err
+	}
+
+	log.Info("imgManifest", "MediaType", imgManifest.MediaType)
+
+	for _, layer := range imgManifest.Layers {
+		log.Info("layers", "digest", layer.Digest.Hex, "type", layer.MediaType)
+	}
+
+	err = installer.pullOCIimage(image, imageName, pullInfo.imageCacheDir, pullInfo.targetDir)
+	if err != nil {
+		log.Info("pullOCIimage", "err", err)
+		return err
+	}
+
+	return nil
+}
+
+func (installer Installer) pullImageInfo(registryAuthPath string, imageName string) (*v1.Image, error) {
+	ref, err := name.ParseReference(imageName)
+	if err != nil {
+		return nil, fmt.Errorf("parsing reference %q: %w", imageName, err)
```

Comment:
we should use `errors.WithMessagef(err, ...)` instead of `fmt.Errorf(...)`  as its better for producing a showing a stacktrace

Reply:
fixed in e278a5f2